Microsoft.Toolkit.Uwp.UI.Controls/CameraPreview/CameraPreview.cs

```diff
@@ -0,0 +1,172 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Helpers.CameraHelper;
+using Windows.Media.Capture.Frames;
+using Windows.Media.Core;
+using Windows.Media.Playback;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Camera Control to preview video. Can subscribe to video frames, software bitmap when they arrive.
+    /// </summary>
+    public partial class CameraPreview : Control, IDisposable
+    {
+        private CameraHelper _cameraHelper;
+        private MediaPlayer _mediaPlayer;
+        private MediaPlayerElement _mediaPlayerElementControl;
+        private Button _toggleFrameSourceGroup;
+        private int _selectedSourceIndex = 0;
+
+        /// <summary>
+        /// Gets Frame Source Groups available for Camera Media Capture.
+        /// </summary>
+        public IReadOnlyList<MediaFrameSourceGroup> FrameSourceGroups { get; private set; }
```

Comment:
I exposed this property based on Nikola's comment, in case the use wants to get the frame source groups available and bind it to a different xaml control.

Reply:
I actualy like that, exposing the entire helper as a read only property. We don't have to have this property and the developer could still access the FrameSourceGroups.